src/main/scala/com/databricks/spark/sql/perf/query.scala

```diff
@@ -19,7 +19,21 @@ package com.databricks.spark.sql.perf
 import org.apache.spark.sql.SQLContext
 import org.apache.spark.sql.catalyst.analysis.UnresolvedRelation
 
-case class Query(name: String, sqlText: String, description: String, collectResults: Boolean)
+/**
+ * The execution mode of a benchmark:
+ * - CollectResults: Benchmark run by collecting queries results
+ *   (e.g. rdd.collect())
+ * - ForeachResults: Benchmark run by iterating through the queries results rows
+ *   (e.g. rdd.foreach(row => Unit))
+ * - WriteParquet(location): Benchmark run by saving the output of each query as a
+ *   parquet file at the specified location
+ */
+abstract class ExecutionMode
+case object CollectResults extends ExecutionMode
```

Comment:
Let's put the actual objects inside of an object like 
```scala
trait ExecutionMode
object ExecutionMode {
  case object CollectResults extends Executionmode
}
```

so we can refer to it by its fully qualified name ExecutionMode.CollectResults from outside of the class, I think this makes the code more readable from the API perspeective.